### PR TITLE
Update zglob to v0.0.4

### PR DESCRIPTION
**Pull request by moskyb**
The version of [zglob](https://github.com/mattn/go-zglob) we're using is positively ancient, and doesn't support useful globbing features like negation or bracketing.

This PR updates our version of go-zglob to the latest, v0.0.4. This update includes both negation and bracketing. Yay!

### CODE REVIEW

File: agent/artifact_uploader_test.go [hunk 1]
```diff
@@ -183,7 +183,7 @@ func TestCollectThatDoesntMatchAnyFiles(t *testing.T) {
 
 	artifacts, err := uploader.Collect()
 	if err != nil {
-		t.Fatalf("uploader.Collect() error = %v", err)
+		t.Fatalf("uploader.Collect() error = %# v", err)
```
Comment (DrJosh9000) [inline agent/artifact_uploader_test.go:186]: Is the diff on this line still needed?

**Review by DrJosh9000 — APPROVED** [review]
🎆

File: agent/artifact_uploader_test.go (hunk 1 above)
Comment (moskyb) [reply agent/artifact_uploader_test.go]: nope! removed :)

**Review by triarius — APPROVED** [review]
Nice 🥇

**Pull request merged**